repair/task_manager_module.hh

```diff
@@ -160,11 +163,46 @@ public:
         return _hints_batchlog_flushed;
     }
 
-    future<> repair_range(const dht::token_range& range, table_id);
+    size_t ranges_size() const noexcept;
+protected:
+    void record_finished_range_repair();
+    future<> run() override;
+
+    virtual future<std::optional<double>> expected_total_workload() const override;
+    virtual std::optional<double> expected_children_number() const override;
+
+    friend class table_repair_task_impl;
+};
+
+class table_repair_task_impl : public repair_task_impl {
+private:
+    shard_repair_task_impl& _local_repair_data;
```

Comment:
Let's call it  _shard_repair_task.  The name _local_repair_data does not give much information.